packages/babel-generator/src/buffer.js

```diff
@@ -99,6 +98,19 @@ export default class Buffer {
   }
 
   endsWith(str: string): boolean {
+    // Fast path to avoid iterating over this._queue.
+    if (str.length === 1) {
+      let last;
+      if (this._queue.length > 0) {
+        const str = this._queue[0][0];
```

Comment:
I think this rule wouldn't hurt too much, unless there are hundreds of places to refactor. So +1 from me.